version: "3"

services:

  redis:
    image: redis
    ports:
      - '6379:6379'

  db:
    restart: always
    image: postgres
    env_file:
      - .env
    environment:
      POSTGRES_DB: ${PG_NAME}
      POSTGRES_USER: ${PG_USER}
      POSTGRES_PASSWORD: ${PG_PASS}
      PG_DATA: /var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d $${PG_NAME} -U $${PG_USER}" ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: bash -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    ports:
      - '8001:8000'
    volumes:
      - .:/code
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO -P eventlet 
    env_file:
      - .env
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO -S django
    env_file:
      - .env
    depends_on:
      - redis
      - app